kart/tabular/working_copy/postgis.py

```diff
@@ -337,3 +343,14 @@ def _apply_meta_schema_json(self, sess, dataset, src_value, dest_value):
             sess.execute(
                 f"""ALTER TABLE {self.table_identifier(table)} ALTER COLUMN {self.quote(col.name)} TYPE {dest_type};"""
             )
+
+    def _check_for_unsupported_ds_types(self, sess, target_datasets, schema="public"):
+        for target_dataset in target_datasets:
+            table = target_datasets[target_dataset]
+            if table.is_spatial:
+                inspector = inspect(sess.bind)
+                if "spatial_ref_sys" not in inspector.get_table_names():
+                    raise InvalidOperation(
+                        "The PostGIS extension is not enabled in the database.\nInstall it with `CREATE EXTENSION postgis;`",
```

Comment:
This message should probably be more clearly related to what the user was trying to do - which is probably either `kart pull` or `kart clone` or `kart checkout mybranch`. They probably configured a postgres WC but may not know what postgis is, so I'd be a bit more direct with this, e.g.:

> Dataset 'roads' requires the postgis extension to be installed in the working copy.
> Install it with `CREATE EXTENSION postgis;`